src/sentry/integrations/jira_server/integration.py

```diff
@@ -318,8 +318,8 @@ def create_webhook(self, external_id, webhook_secret, install, credentials):
                 extra={"error": six.text_type(err), "external_id": external_id},
             )
             try:
-                details = err.json["messages"][0].values().pop()
-            except Exception:
+                details = list(err.json["messages"][0].values()).pop()
```

Comment:
I don't understand the original logic here. Since dicts are unordered, doing `values().pop()` means "just pick something" to me. If that's the case, why not do the following, which should be more efficient
```suggestion
                details = next(x for x in err.json["messages"][0].values())
```

Reply:
Pretty sure this will have an `IteratorStop` error right?

Reply:
If it's not in there.

Reply:
Yes, that's right. With `.pop()` you get `IndexError: pop from empty list` instead.

Reply:
Alright, switched this over.